integration/ruler_test.go

```diff
@@ -1739,6 +1739,106 @@ func TestRulerEvalWithQueryFrontend(t *testing.T) {
 	}
 }
```

Comment:
Can we try to cover group labels in the API backup test case? Just to make sure it works for that scenario

`TestRulerAPIShardingWithAPIRulesBackupEnabled`